addons/website/static/src/components/configurator/configurator.js

```diff
@@ -28,27 +26,72 @@ const WEBSITE_PURPOSES = {
 };
 
-const hex2lab = (hex) => {
-    let r = parseInt(hex.substring(1, 3), 16) / 255, g = parseInt(hex.substring(3, 5), 16) / 255, b = parseInt(hex.substring(5, 7), 16) / 255, x, y, z;
-    [r, g, b] = [r, g, b].map((x) => (x > 0.04045) ? Math.pow((x + 0.055) / 1.055, 2.4) : x / 12.92);
-    x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047;
-    y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000;
-    z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883;
-    [x, y, z] = [x, y, z].map((x) => (x > 0.008856) ? Math.pow(x, 1 / 3) : (7.787 * x) + 16 / 116);
-    return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)];
-};
+const PALETTE_NAMES = [
+    'bookstore-5',
+    'beauty-3',
+    'cobalt-1',
+    'generic-6',
+    'enark-1',
+    'bewise-1',
+    'generic-16',
+    'generic-12',
+    'clean-2',
+    'generic-8',
+    'avantgarde-3',
+    'nano-2',
+    'graphene-2',
+    'notes-3',
+    'generic-2',
+    'avantgarde-6',
+];
+
+//---------------------------------------------------------
+// Utils
+//---------------------------------------------------------
+
+/**
+ * Return an array containing the red, green and blue components
+ * of a hex color in decimal base.
+ *
+ * @param {string} color
+ * @returns {number[]}
+ */
+function getRGBDecimal(color) {
+    const rgb = [1, 3, 5].map((idx) => parseInt(color.slice(idx, idx+2), 16))
+    return rgb;
+}
+
+/**
+ * Weighted average of two numbers returned in hexadecimal base
+ * as a 2 characters string.
+ *
+ * @param {number} v1
+ * @param {number} v2
+ * @param {number} weight
+ * @returns {string}
+ */
+function mixValue(v1, v2, weight) {
+    const mixedValue = Math.floor(v2 + (v1 - v2) * weight).toString(16);
+    if (mixedValue.length === 1) {
+        return '0' + mixedValue;
+    }
+    return mixedValue;
+}
 
-const deltaE = (hexA, hexB) => {
-    let labA = hex2lab(hexA), labB = hex2lab(hexB);
-    let deltaL = labA[0] - labB[0], deltaA = labA[1] - labB[1], deltaB = labA[2] - labB[2];
-    let c1 = Math.sqrt(labA[1] * labA[1] + labA[2] * labA[2]), c2 = Math.sqrt(labB[1] * labB[1] + labB[2] * labB[2]);
-    let deltaC = c1 - c2;
-    let deltaH = deltaA * deltaA + deltaB * deltaB - deltaC * deltaC;
-    deltaH = deltaH < 0 ? 0 : Math.sqrt(deltaH);
-    let sc = 1.0 + 0.045 * c1, sh = 1.0 + 0.015 * c1;
-    let deltaLKlsl = deltaL / (1.0), deltaCkcsc = deltaC / (sc), deltaHkhsh = deltaH / (sh);
-    let i = deltaLKlsl * deltaLKlsl + deltaCkcsc * deltaCkcsc + deltaHkhsh * deltaHkhsh;
-    return i < 0 ? 0 : Math.sqrt(i);
+/**
+ * Mix two colors by applying a weighted average on
+ * the red, green and blue components.
+ *
+ * @param {string} color1
+ * @param {string} color2
+ * @param {number} weight
+ * @returns {string}
+ */
+function mix(color1, color2, weight) {
+    const rgb1 = getRGBDecimal(color1), rgb2 = getRGBDecimal(color2);;
+    const rgb = rgb1.map((v, idx) => mixValue(v, rgb2[idx], weight));
+    const color = '#' + rgb.join('');
+    return color;
 };
```

Comment:
Maybe those should be moved next to the already-existing color utils as well?

Reply:
in colorpicker.js or is there an other color utils file?

Reply:
Yes, colorpicker.js I guess